src/scripts/notificationblock.js

```diff
@@ -79,9 +80,12 @@ const onButtonClicked = async function ({ currentTarget }) {
   });
 };
 
-const blockPostFilter = async ({ id, canEdit, rebloggedRootId }) => {
+const blockPostFilter = async ({ blogName, rebloggedRootName, rebloggedFromName, id, rebloggedRootId }) => {
   const rootId = rebloggedRootId || id;
-  return canEdit && blockedPostTargetIDs.includes(rootId) === false;
+  const yourOriginalPost = userBlogNames.includes(rebloggedRootName);
+  const yourReblog = userBlogNames.includes(blogName) || userBlogNames.includes(rebloggedFromName);
+
+  return (yourOriginalPost || yourReblog) && blockedPostTargetIDs.includes(rootId) === false;
```

Comment:
These names aren't strictly accurate... `rebloggedRootName` will be undefined when viewing _any_ original post, which means all original posts when viewing your own blog would be counted as `yourReblog` rather than `yourOriginalPost`.

What's being tested for *is* correct though, so let's just combine the two constants.

```suggestion
  const canReceiveActivity = userBlogNames.includes(blogName) ||
    userBlogNames.includes(rebloggedFromName) ||
    userBlogNames.includes(rebloggedRootName);

  return canReceiveActivity && blockedPostTargetIDs.includes(rootId) === false;
```

Reply:
Oh, right! Whoops. Sometimes I think I need to just write documentation for the timelineObject fields for myself as if they were official API endpoints :D